File: Entregable 3/source/c/Gramatica.c
/*
 * Gramatica.c
 *
 * 2025/05/27 09:24:37
 *
 * Archivo generado por GikGram 2.0
 *
 * Copyright © Olminsky 2011 Derechos reservados
 * Reproducción sin fines de lucro permitida
 */
#include "Gramatica.h"

/* Tabla de parsing */
const int TablaParsing[6][NO_TERMINAL_INICIAL] =
{
	/* <inicio> */ {0,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1},
	/* <inicio2> */ {-1,1,1,1,1,1,1,1,-1,-1,-1},
	/* <seccion> */ {-1,2,3,4,5,6,7,7,-1,-1,-1},
	/* <punto_entrada> */ {-1,-1,-1,-1,-1,-1,8,9,-1,-1,-1},
	/* <identificador> */ {-1,-1,-1,-1,-1,-1,-1,-1,-1,10,-1},
	/* <final> */ {-1,-1,-1,-1,-1,-1,-1,11,-1,-1,-1}
};

/* Tabla de lados derechos */
const int LadosDerechos[12][MAX_LADO_DER] =
{
	{12,8,15,0},
	{16,14,13,-1},
	{1,-1,-1,-1},
	{2,-1,-1,-1},
	{3,-1,-1,-1},
	{4,-1,-1,-1},
	{5,-1,-1,-1},
	{-1,-1,-1,-1},
	{6,-1,-1,-1},
	{-1,-1,-1,-1},
	{9,-1,-1,-1},
	{7,-1,-1,-1}
};

File: Entregable 3/Prueba con el gikgram/source/java/Gramatica/Gramatica.java
/*
 * Gramatica.java
 *
 * 2025/05/29 10:22:23
 *
 * Archivo generado por GikGram 2.0
 *
 * Copyright © Olminsky 2011 Derechos reservados
 * Reproducción sin fines de lucro permitida
 */

package Gramatica;

/**
 * Esta clase contiene:
 * - Constantes necesarias para el driver de parsing
 * - Constantes con las rutinas semánticas
 * - Y los métodos necesarios para el driver de parsing
 */
public abstract class Gramatica
{
	/* Esta es la única clase que se accede fuera del paquete Gramatica */

	/**
	 * Constante que contiene el código de familia del terminal de fin de archivo
	 */
	public static final int MARCA_DERECHA = 3;

	/**
	 * Constante que contiene el número del no-terminal inicial
	 * (el primer no-terminal que aparece en la gramática)
	 */
	public static final int NO_TERMINAL_INICIAL = 4;

	/**
	 * Constante que contiene el número máximo de columnas que tiene los lados derechos
	 */
	public static final int MAX_LADO_DER = 5;

	/**
	 * Constante que contiene el número máximo de follows
	 */
	public static final int MAX_FOLLOWS = 2;

	/* Constantes con las rutinas semánticas */
	public static final int InicioTSG = 6;
	public static final int otro = 7;

	/**
	 * Método esTerminal
			Devuelve true si el símbolo es un terminal
			o false de lo contrario
	 * @param numSimbolo
			Número de símbolo
	 */
	public static final boolean esTerminal(int numSimbolo)
	{
		return ((0 <= numSimbolo) && (numSimbolo <= 3));
	}

	/**
	 * Método esNoTerminal
			Devuelve true si el símbolo es un no-terminal
			o false de lo contrario
	 * @param numSimbolo
			Número de símbolo
	 */
	public static final boolean esNoTerminal(int numSimbolo)
	{
		return ((4 <= numSimbolo) && (numSimbolo <= 5));
	}

	/**
	 * Método esSimboloSemantico
			Devuelve true si el símbolo es un símbolo semántico
			(incluyendo los símbolos de generación de código)
			o false de lo contrario
	 * @param numSimbolo
			Número de símbolo
	 */
	public static final boolean esSimboloSemantico(int numSimbolo)
	{
		return ((6 <= numSimbolo) && (numSimbolo <= 7));
	}

	/**
	 * Método getTablaParsing
			Devuelve el número de regla contenida en la tabla de parsing
	 * @param numNoTerminal
			Número del no-terminal
	 * @param numTerminal
			Número del terminal
	 */
	public static final int getTablaParsing(int numNoTerminal, int numTerminal)
	{
		return GTablaParsing.getTablaParsing(numNoTerminal, numTerminal);
	}

	/**
	 * Método getLadosDerechos
			Obtiene un símbolo del lado derecho de la regla
	 * @param numRegla
			Número de regla
	 * @param numColumna
			Número de columna
	 */
	public static final int getLadosDerechos(int numRegla, int numColumna)
	{
		return GLadosDerechos.getLadosDerechos(numRegla, numColumna);
	}

	/**
	 * Método getNombresTerminales
			Obtiene el nombre del terminal
	 * @param numTerminal
			Número del terminal
	 */
	public static final String getNombresTerminales(int numTerminal)
	{
		return GNombresTerminales.getNombresTerminales(numTerminal);
	}

	/**
	 * Método getTablaFollows
			Obtiene el número de terminal del follow del no-terminal
	 * @param numNoTerminal
			Número de no-terminal
	 * @param numColumna
			Número de columna
	 */
	public static final int getTablaFollows(int numNoTerminal, int numColumna)
	{
		return GTablaFollows.getTablaFollows(numNoTerminal, numColumna);
	}
}
